docs/how-to/purge-data-on-shadow-table.rst

```diff
@@ -0,0 +1,30 @@
+==========================================
+Purge data on shadow tables
+==========================================
+
+The purge steps is disable by default and plans to run before proceeding to actual upgrade steps.
+This can be turn on with the `--purge`` flag.
+
+This purge step is a performance optimisation, delete data from the shadow tables in nova database. The behavior is equal to run juju action `purge` on nova-cloud-controller unit, which can help to reduce the size of the db.
+
+The `purge-before-date` flag is supported to delete the data older than the date provided. The date string format should be YYYY-MM-DD[HH:mm][:ss]. This flag is useful to retain recent data for debugging or data retention policies.
+
+
+Usage examples
+--------------
+
+.. terminal::
+   :input cou plan --purge --purge-before-date 2000-01-02
+
+
+More information
+----------------
+
+- `nova-cloud-controller charm actions`_
+- `nova-manage reference`_ - see `purge` subcommand
+- :doc:`Archive old data <how-to/archive-old-data>`
+- :doc:`Nova data migration <explanation/nova-data-migration>`
+
+.. LINKS
+.. _nova-cloud-controller charm actions: https://charmhub.io/nova-cloud-controller/actions
+.. _nova-manage reference: https://docs.openstack.org/nova/rocky/cli/nova-manage.html
```

Comment:
Important information that's missing in these docs is the benefits of archive + purge:

- performance optimisation for during the upgrade process (albeit small)
- reduce database size and disk usage
- improve performance of database queries during cloud operation because the db is smaller

I think this how-to is actually a good place for this information: you already mention performance optimisation, and I think it's important for a doc to include the 'why'.

Reply:
I opened https://github.com/canonical/charmed-openstack-upgrader/issues/483 to track this.